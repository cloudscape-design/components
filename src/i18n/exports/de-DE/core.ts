// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import Alert from '../../messages/de-DE/alert';
import Autosuggest from '../../messages/de-DE/autosuggest';
import Badge from '../../messages/de-DE/badge';
import Box from '../../messages/de-DE/box';
import ButtonDropdown from '../../messages/de-DE/button-dropdown';
import Button from '../../messages/de-DE/button';
import Checkbox from '../../messages/de-DE/checkbox';
import ColumnLayout from '../../messages/de-DE/column-layout';
import Container from '../../messages/de-DE/container';
import ExpandableSection from '../../messages/de-DE/expandable-section';
import Flashbar from '../../messages/de-DE/flashbar';
import FormField from '../../messages/de-DE/form-field';
import Form from '../../messages/de-DE/form';
import Grid from '../../messages/de-DE/grid';
import Header from '../../messages/de-DE/header';
import Icon from '../../messages/de-DE/icon';
import Input from '../../messages/de-DE/input';
import Link from '../../messages/de-DE/link';
import Modal from '../../messages/de-DE/modal';
import Multiselect from '../../messages/de-DE/multiselect';
import Popover from '../../messages/de-DE/popover';
import ProgressBar from '../../messages/de-DE/progress-bar';
import RadioGroup from '../../messages/de-DE/radio-group';
import SegmentedControl from '../../messages/de-DE/segmented-control';
import Select from '../../messages/de-DE/select';
import SpaceBetween from '../../messages/de-DE/space-between';
import Spinner from '../../messages/de-DE/spinner';
import StatusIndicator from '../../messages/de-DE/status-indicator';
import Tabs from '../../messages/de-DE/tabs';
import TextContent from '../../messages/de-DE/text-content';
import Textarea from '../../messages/de-DE/textarea';
import Tiles from '../../messages/de-DE/tiles';
import Toggle from '../../messages/de-DE/toggle';
import TokenGroup from '../../messages/de-DE/token-group';

export default {
  alert: Alert,
  autosuggest: Autosuggest,
  badge: Badge,
  box: Box,
  'button-dropdown': ButtonDropdown,
  button: Button,
  checkbox: Checkbox,
  'column-layout': ColumnLayout,
  container: Container,
  'expandable-section': ExpandableSection,
  flashbar: Flashbar,
  'form-field': FormField,
  form: Form,
  grid: Grid,
  header: Header,
  icon: Icon,
  input: Input,
  link: Link,
  modal: Modal,
  multiselect: Multiselect,
  popover: Popover,
  'progress-bar': ProgressBar,
  'radio-group': RadioGroup,
  'segmented-control': SegmentedControl,
  select: Select,
  'space-between': SpaceBetween,
  spinner: Spinner,
  'status-indicator': StatusIndicator,
  tabs: Tabs,
  'text-content': TextContent,
  textarea: Textarea,
  tiles: Tiles,
  toggle: Toggle,
  'token-group': TokenGroup,
};
