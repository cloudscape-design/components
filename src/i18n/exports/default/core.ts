// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import Alert from '../../messages/default/alert';
import Autosuggest from '../../messages/default/autosuggest';
import Badge from '../../messages/default/badge';
import Box from '../../messages/default/box';
import ButtonDropdown from '../../messages/default/button-dropdown';
import Button from '../../messages/default/button';
import Checkbox from '../../messages/default/checkbox';
import ColumnLayout from '../../messages/default/column-layout';
import Container from '../../messages/default/container';
import ExpandableSection from '../../messages/default/expandable-section';
import Flashbar from '../../messages/default/flashbar';
import FormField from '../../messages/default/form-field';
import Form from '../../messages/default/form';
import Grid from '../../messages/default/grid';
import Header from '../../messages/default/header';
import Icon from '../../messages/default/icon';
import Input from '../../messages/default/input';
import Link from '../../messages/default/link';
import Modal from '../../messages/default/modal';
import Multiselect from '../../messages/default/multiselect';
import Popover from '../../messages/default/popover';
import ProgressBar from '../../messages/default/progress-bar';
import RadioGroup from '../../messages/default/radio-group';
import SegmentedControl from '../../messages/default/segmented-control';
import Select from '../../messages/default/select';
import SpaceBetween from '../../messages/default/space-between';
import Spinner from '../../messages/default/spinner';
import StatusIndicator from '../../messages/default/status-indicator';
import Tabs from '../../messages/default/tabs';
import TextContent from '../../messages/default/text-content';
import Textarea from '../../messages/default/textarea';
import Tiles from '../../messages/default/tiles';
import Toggle from '../../messages/default/toggle';
import TokenGroup from '../../messages/default/token-group';

export default {
  alert: Alert,
  autosuggest: Autosuggest,
  badge: Badge,
  box: Box,
  'button-dropdown': ButtonDropdown,
  button: Button,
  checkbox: Checkbox,
  'column-layout': ColumnLayout,
  container: Container,
  'expandable-section': ExpandableSection,
  flashbar: Flashbar,
  'form-field': FormField,
  form: Form,
  grid: Grid,
  header: Header,
  icon: Icon,
  input: Input,
  link: Link,
  modal: Modal,
  multiselect: Multiselect,
  popover: Popover,
  'progress-bar': ProgressBar,
  'radio-group': RadioGroup,
  'segmented-control': SegmentedControl,
  select: Select,
  'space-between': SpaceBetween,
  spinner: Spinner,
  'status-indicator': StatusIndicator,
  tabs: Tabs,
  'text-content': TextContent,
  textarea: Textarea,
  tiles: Tiles,
  toggle: Toggle,
  'token-group': TokenGroup,
};
