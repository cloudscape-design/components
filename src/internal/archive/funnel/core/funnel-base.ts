// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { AnalyticsMetadata } from '../../interfaces';
import { CallbackFunction, ErrorDetails, FunnelBaseStatus, FunnelType, Observer } from '../types';
import { getUuid } from '../utils';

export abstract class FunnelBase<TStatus extends string = FunnelBaseStatus> {
  protected status: TStatus[] = [];
  protected observers: Observer[] = [];

  public id: string;
  public type?: FunnelType;
  public name?: string;
  public metadata?: AnalyticsMetadata;

  constructor(initialStatus: TStatus) {
    this.id = getUuid();
    this.status = [initialStatus];
  }

  protected setStatus(newStatus: TStatus, callback?: CallbackFunction): void {
    if (this.getStatus() !== newStatus) {
      this.status.push(newStatus);
      this.notifyObservers();

      callback?.();
    }
  }

  setName(name: string): void {
    this.name = name;
    this.notifyObservers();
  }

  setFunnelType(type: FunnelType): void {
    this.type = type;
    this.notifyObservers();
  }

  setMetadata(metadata?: AnalyticsMetadata): void {
    if (this.metadata === metadata) {
      return;
    }

    this.metadata = metadata;
    this.notifyObservers();
  }

  getStatus(): TStatus {
    if (this.status.length === 0) {
      return 'unknown' as TStatus;
    }
    return this.status[this.status.length - 1];
  }

  getPreviousStatus(): TStatus {
    if (this.status.length < 2) {
      return 'unknown' as TStatus;
    }

    return this.status[this.status.length - 2];
  }

  start(callback?: CallbackFunction) {
    if (this.getStatus() === 'started') {
      return;
    }

    this.setStatus('started' as TStatus);
    callback?.();
  }

  complete(callback?: CallbackFunction) {
    if (this.getStatus() === 'completed') {
      return;
    }

    this.setStatus('completed' as TStatus);
    callback?.();
  }

  error(details: ErrorDetails, callback?: CallbackFunction) {
    if (!details.errorText) {
      return;
    }

    this.setStatus('error' as TStatus);
    callback?.();
  }

  addObserver(observer: Observer): void {
    this.observers.push(observer);
  }

  removeObserver(observer: Observer): void {
    this.observers = this.observers.filter(obs => obs !== observer);
  }

  protected notifyObservers(): void {
    for (const observer of this.observers) {
      observer.update(this);
    }
  }
}
