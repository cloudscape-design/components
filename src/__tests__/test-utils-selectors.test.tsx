// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { NodePath, PluginObj, transformSync, types } from '@babel/core';
import fs from 'fs';
import glob from 'glob';
import { flatten, zip } from 'lodash';
import path from 'path';

const defaultPlugins = [require('@babel/plugin-syntax-typescript')] as const;

// The test extracts generated selectors from the compiled output and matches those against the snapshot.
test('test-utils selectors', () => {
  // Find referenced selector files and properties.
  const selectorsFilePathToUsedProperties = new Map<string, Set<string>>();
  for (const file of glob.sync('src/test-utils/selectors/**/*.ts')) {
    extractSelectorProperties(file, (filePath, propertyKey) => {
      const properties = selectorsFilePathToUsedProperties.get(filePath) ?? new Set();
      properties.add(propertyKey);
      selectorsFilePathToUsedProperties.set(filePath, properties);
    });
  }

  // Find referenced selector values.
  const componentToSelectors: Record<string, string[]> = {};
  for (const [filePath, properties] of selectorsFilePathToUsedProperties) {
    extractComponentSelectors(filePath, [...properties], selector => {
      const componentName = getComponentNameFromFilePath(filePath);
      componentToSelectors[componentName] = [...(componentToSelectors[componentName] ?? []), selector].sort();
    });
  }
  expect(componentToSelectors).toMatchSnapshot();
});

function extractSelectorProperties(file: string, onExtract: (filePath: string, propertyKey: string) => void) {
  function getPropertyName(path: NodePath<types.MemberExpression>) {
    if (path.node.property.type === 'Identifier') {
      return path.node.property.name;
    } else if (path.node.property.type === 'StringLiteral') {
      return path.node.property.value;
    } else if (path.node.property.type === 'TemplateLiteral') {
      return buildTemplateString(path.node.property);
    } else {
      throw new Error(`Unhandled selector access type at ${file}:${path.node.loc?.start.line}.`);
    }
  }

  // Build string literal from template string replacing arguments with wildcards ("*").
  // For example, a template string `flash-type-${statusType}` becomes "flash-type-*".
  function buildTemplateString(node: types.TemplateLiteral) {
    let literal = '';
    for (const element of flatten(zip(node.quasis, node.expressions))) {
      if (!element) {
        continue;
      } else if (element.type === 'TemplateElement') {
        literal += element.value.raw;
      } else if (element.type === 'Identifier') {
        literal += '*';
      } else if (element.type === 'NumericLiteral') {
        literal += element.value;
      } else {
        throw new Error(`Unhandled template literal type at ${file}:${node.loc?.start.line}.`);
      }
    }
    return literal;
  }

  function extractor(): PluginObj {
    return {
      visitor: {
        // Find selector references and extract used property names.
        MemberExpression(path: NodePath<types.MemberExpression>) {
          if (path.node.object.type === 'Identifier') {
            const binding = path.scope.getBinding(path.node.object.name);
            if (
              binding &&
              binding.kind === 'module' &&
              binding.path.parent.type === 'ImportDeclaration' &&
              binding.path.parent.source.value.endsWith('selectors.js')
            ) {
              onExtract(resolveSelectorsPath(binding.path.parent.source.value), getPropertyName(path));
            }
          }
        },
      },
    } as PluginObj;
  }
  const source = fs.readFileSync(file, 'utf-8');
  transformSync(source, { babelrc: false, configFile: false, plugins: [...defaultPlugins, extractor] })?.code;
}

function extractComponentSelectors(file: string, usedProperties: string[], onExtract: (selector: string) => void) {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const source = require(file);
  if (typeof source.default !== 'object') {
    throw new Error(`Unexpected selectors file format at ${file}.`);
  }
  for (const [property, value] of Object.entries(source.default)) {
    if (typeof value !== 'string') {
      throw new Error(`Unexpected selectors file format at ${file}, "${property}".`);
    }
    if (matchProperties(usedProperties, property)) {
      onExtract(trimSelectorHash(value as string));
    }
  }
}

function trimSelectorHash(selector: string) {
  const splitSelector = selector.replace('.', '').split('_');
  if (splitSelector.length >= 5) {
    splitSelector.splice(splitSelector.length - 2, splitSelector.length);
    return splitSelector.join('_');
  }
  return selector;
}

function resolveSelectorsPath(importPath: string) {
  let selectorsFile = importPath;
  // Path to selectors file from 'lib/components'.
  selectorsFile = path.resolve(path.relative('lib/components/test-utils', importPath));
  // Absolute path to selectors file.
  selectorsFile = path.resolve(path.join('lib/components', selectorsFile));
  return selectorsFile;
}

function getComponentNameFromFilePath(filePath: string) {
  return filePath.match(/lib\/components\/([\w-]+)/)![1];
}

// Match property against the used properties supporting wildcards ("*").
// For example, property "flash-type-*" matches "flash-type-error", "flash-type-in-progress", and so on.
function matchProperties(usedProperties: string[], property: string) {
  for (const testProperty of usedProperties) {
    if (testProperty === property) {
      return true;
    }
    if (matchWildcard(testProperty, property)) {
      return true;
    }
  }
  return false;
}

function matchWildcard(testProperty: string, propertyToMatch: string) {
  if (testProperty.includes('*')) {
    testProperty = testProperty.replace(/\*/g, '.*');
    return new RegExp('^' + testProperty + '$').test(propertyToMatch);
  }
  return false;
}
