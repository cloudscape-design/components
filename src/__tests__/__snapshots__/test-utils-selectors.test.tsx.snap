// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-utils selectors 1`] = `
Object {
  "alert": Array [
    "awsui_action-button_mx3cw",
    "awsui_action-slot_mx3cw",
    "awsui_alert_mx3cw",
    "awsui_content_mx3cw",
    "awsui_dismiss-button_mx3cw",
    "awsui_header_mx3cw",
    "awsui_root_mx3cw",
  ],
  "anchor-navigation": Array [
    "awsui_anchor-item--active_17oho",
    "awsui_anchor-link-info_17oho",
    "awsui_anchor-link-text_17oho",
    "awsui_anchor-link_17oho",
    "awsui_anchor-list_17oho",
    "awsui_root_17oho",
  ],
  "annotation-context": Array [
    "awsui_annotation_1hpp3",
    "awsui_content_1hpp3",
    "awsui_finish-button_1hpp3",
    "awsui_header_1hpp3",
    "awsui_hotspot_1hpp3",
    "awsui_next-button_1hpp3",
    "awsui_previous-button_1hpp3",
    "awsui_step-counter-content_1hpp3",
  ],
  "app-layout": Array [
    "awsui_active-drawer-close-button_1fj9k",
    "awsui_active-drawer_1fj9k",
    "awsui_breadcrumbs_1fj9k",
    "awsui_content_1fj9k",
    "awsui_drawers-slider_1fj9k",
    "awsui_drawers-trigger_1fj9k",
    "awsui_navigation-close_1fj9k",
    "awsui_navigation-toggle_1fj9k",
    "awsui_navigation_1fj9k",
    "awsui_notifications_1fj9k",
    "awsui_overflow-menu_1fj9k",
    "awsui_root_1fj9k",
    "awsui_tools-close_1fj9k",
    "awsui_tools-toggle_1fj9k",
    "awsui_tools_1fj9k",
  ],
  "area-chart": Array [
    "awsui_root_y1yrh",
    "awsui_series--highlighted_y1yrh",
    "awsui_series_y1yrh",
  ],
  "attribute-editor": Array [
    "awsui_add-button_n4qlp",
    "awsui_additional-info_n4qlp",
    "awsui_empty_n4qlp",
    "awsui_field_n4qlp",
    "awsui_remove-button_n4qlp",
    "awsui_root_n4qlp",
    "awsui_row-control_n4qlp",
    "awsui_row_n4qlp",
  ],
  "autosuggest": Array [
    "awsui_root_12hwc",
  ],
  "badge": Array [
    "awsui_badge_1yjyg",
  ],
  "bar-chart": Array [
    "awsui_root_1gfe1",
  ],
  "box": Array [
    "awsui_root_18wu0",
  ],
  "breadcrumb-group": Array [
    "awsui_anchor_1kosq",
    "awsui_breadcrumb-group_d19fg",
    "awsui_breadcrumb_1kosq",
    "awsui_item_d19fg",
  ],
  "button": Array [
    "awsui_button_vjswe",
    "awsui_content_vjswe",
    "awsui_icon-left_vjswe",
  ],
  "button-dropdown": Array [
    "awsui_button-dropdown_sne0l",
    "awsui_description_sne0l",
    "awsui_dropdown-trigger_sne0l",
    "awsui_expandable_16mm3",
    "awsui_highlighted_93a1u",
    "awsui_item-element_93a1u",
    "awsui_split-trigger_sne0l",
    "awsui_test-utils-button-trigger_sne0l",
    "awsui_title_sne0l",
  ],
  "calendar": Array [
    "awsui_calendar-date-selected_1ykar",
    "awsui_calendar-date_1ykar",
    "awsui_calendar-header_1ykar",
    "awsui_calendar-next-btn_1ykar",
    "awsui_calendar-prev-btn_1ykar",
    "awsui_calendar-row_1ykar",
    "awsui_root_1ykar",
  ],
  "cards": Array [
    "awsui_card-header-inner_p8a6i",
    "awsui_card-selected_p8a6i",
    "awsui_card_p8a6i",
    "awsui_empty_p8a6i",
    "awsui_loading_p8a6i",
    "awsui_root_p8a6i",
    "awsui_section-content_p8a6i",
    "awsui_section-header_p8a6i",
    "awsui_section_p8a6i",
    "awsui_selection-control_p8a6i",
  ],
  "checkbox": Array [
    "awsui_root_k2y2q",
  ],
  "code-editor": Array [
    "awsui_code-editor_1gl9c",
    "awsui_editor_1gl9c",
    "awsui_error-screen_1gl9c",
    "awsui_loading-screen_1gl9c",
    "awsui_pane_1gl9c",
    "awsui_status-bar_1gl9c",
    "awsui_status-bar__cog-button_1gl9c",
    "awsui_tab-button--errors_1gl9c",
    "awsui_tab-button--warnings_1gl9c",
  ],
  "collection-preferences": Array [
    "awsui_cancel-button_tc96w",
    "awsui_confirm-button_tc96w",
    "awsui_content-density_tc96w",
    "awsui_content-display-description_tc96w",
    "awsui_content-display-option-content_tc96w",
    "awsui_content-display-option-label_tc96w",
    "awsui_content-display-option-list_tc96w",
    "awsui_content-display-option-toggle_tc96w",
    "awsui_content-display-option_tc96w",
    "awsui_content-display-title_tc96w",
    "awsui_content-display_tc96w",
    "awsui_custom_tc96w",
    "awsui_modal-root_tc96w",
    "awsui_page-size-form-field_tc96w",
    "awsui_page-size-radio-group_tc96w",
    "awsui_page-size_tc96w",
    "awsui_root_tc96w",
    "awsui_sticky-columns-first_tc96w",
    "awsui_sticky-columns-form-field_tc96w",
    "awsui_sticky-columns-last_tc96w",
    "awsui_sticky-columns-radio-group_tc96w",
    "awsui_striped-rows_tc96w",
    "awsui_trigger-button_tc96w",
    "awsui_visible-content-group-label_tc96w",
    "awsui_visible-content-group_tc96w",
    "awsui_visible-content-groups_tc96w",
    "awsui_visible-content-option-label_tc96w",
    "awsui_visible-content-option_tc96w",
    "awsui_visible-content-title_tc96w",
    "awsui_visible-content-toggle_tc96w",
    "awsui_visible-content_tc96w",
    "awsui_wrap-lines_tc96w",
  ],
  "column-layout": Array [
    "awsui_column-layout_vvxn7",
  ],
  "container": Array [
    "awsui_content-wrapper_14iqq",
    "awsui_content_14iqq",
    "awsui_footer_14iqq",
    "awsui_header_14iqq",
    "awsui_media_14iqq",
    "awsui_root_14iqq",
  ],
  "content-layout": Array [
    "awsui_content_5gtk3",
    "awsui_header_5gtk3",
    "awsui_layout_5gtk3",
  ],
  "copy-to-clipboard": Array [
    "awsui_root_ljpwc",
    "awsui_text-to-copy_ljpwc",
  ],
  "date-input": Array [
    "awsui_root_yodkx",
  ],
  "date-picker": Array [
    "awsui_open-calendar-button_145pa",
    "awsui_root_145pa",
  ],
  "date-range-picker": Array [
    "awsui_apply-button_mgja0",
    "awsui_calendar-header_mgja0",
    "awsui_calendar-next-month-btn_mgja0",
    "awsui_calendar-prev-month-btn_mgja0",
    "awsui_cancel-button_mgja0",
    "awsui_clear-button_mgja0",
    "awsui_custom-range-duration-input_16zmw",
    "awsui_custom-range-unit-select_16zmw",
    "awsui_day_1mfbn",
    "awsui_dropdown_mgja0",
    "awsui_end-date-input_mgja0",
    "awsui_end-date_1mfbn",
    "awsui_end-time-input_mgja0",
    "awsui_first-grid_mgja0",
    "awsui_label_mgja0",
    "awsui_mode-switch_mgja0",
    "awsui_relative-range-radio-group_16zmw",
    "awsui_root_mgja0",
    "awsui_second-grid_mgja0",
    "awsui_start-date-input_mgja0",
    "awsui_start-date_1mfbn",
    "awsui_start-time-input_mgja0",
    "awsui_validation-error_mgja0",
    "awsui_week_1mfbn",
  ],
  "drawer": Array [
    "awsui_drawer_1sxt8",
    "awsui_header_1sxt8",
    "awsui_test-utils-drawer-content_1sxt8",
  ],
  "expandable-section": Array [
    "awsui_content-expanded_gwq0h",
    "awsui_content_gwq0h",
    "awsui_expand-button_gwq0h",
    "awsui_header-text_gwq0h",
    "awsui_header_gwq0h",
    "awsui_icon-container_gwq0h",
    "awsui_root_gwq0h",
  ],
  "file-upload": Array [
    "awsui_file-option-last-modified_ezgb4",
    "awsui_file-option-name_ezgb4",
    "awsui_file-option-size_ezgb4",
    "awsui_file-option-thumbnail-image_ezgb4",
    "awsui_hints_1ubbm",
    "awsui_root_1ubbm",
    "awsui_upload-button_4xu1k",
    "awsui_upload-input_4xu1k",
  ],
  "flashbar": Array [
    "awsui_action-button_1q84n",
    "awsui_action-slot_1q84n",
    "awsui_dismiss-button_1q84n",
    "awsui_flash-content_1q84n",
    "awsui_flash-header_1q84n",
    "awsui_flash-list-item_1q84n",
    "awsui_flash-type-error_1q84n",
    "awsui_flash-type-in-progress_1q84n",
    "awsui_flash-type-info_1q84n",
    "awsui_flash-type-success_1q84n",
    "awsui_flash-type-warning_1q84n",
    "awsui_flashbar_1q84n",
    "awsui_notification-bar_1q84n",
  ],
  "form": Array [
    "awsui_actions_1i0s3",
    "awsui_content_1i0s3",
    "awsui_error_1i0s3",
    "awsui_header_1i0s3",
    "awsui_root_1i0s3",
    "awsui_secondary-actions_1i0s3",
  ],
  "form-field": Array [
    "awsui_constraint_14mhv",
    "awsui_control_14mhv",
    "awsui_description_14mhv",
    "awsui_error_14mhv",
    "awsui_error__message_14mhv",
    "awsui_hints_14mhv",
    "awsui_info_14mhv",
    "awsui_label_14mhv",
    "awsui_root_14mhv",
    "awsui_secondary-control_14mhv",
  ],
  "grid": Array [
    "awsui_grid-column_14yj0",
    "awsui_grid_14yj0",
  ],
  "header": Array [
    "awsui_actions_2qdw9",
    "awsui_counter_2qdw9",
    "awsui_description_2qdw9",
    "awsui_heading-text_2qdw9",
    "awsui_info_2qdw9",
    "awsui_root_2qdw9",
  ],
  "help-panel": Array [
    "awsui_content_1d237",
    "awsui_footer_1d237",
    "awsui_header_1d237",
    "awsui_help-panel_1d237",
  ],
  "hotspot": Array [
    "awsui_root_cqcqk",
  ],
  "icon": Array [
    "awsui_icon_h11ix",
  ],
  "input": Array [
    "awsui_input-button-right_2rhyz",
    "awsui_input-container_2rhyz",
    "awsui_input_2rhyz",
    "awsui_root_2rhyz",
  ],
  "internal": Array [
    "awsui_application_1fcus",
    "awsui_axis-label--x_f0fot",
    "awsui_axis-label--y_f0fot",
    "awsui_button-trigger_18eso",
    "awsui_button_m5h9f",
    "awsui_chart-filter_1px7g",
    "awsui_control_1wepg",
    "awsui_description_1p2cx",
    "awsui_description_1wepg",
    "awsui_disabled_15o6u",
    "awsui_dismiss-button_10m3l",
    "awsui_dropdown_qwoo0",
    "awsui_filter-container_z5mul",
    "awsui_filtering-match-highlight_1p2cx",
    "awsui_handle_sdha6",
    "awsui_has-background_15o6u",
    "awsui_highlighted_15o6u",
    "awsui_inner-list-item_10ipo",
    "awsui_key_10ipo",
    "awsui_label-tag_1p2cx",
    "awsui_label-wrapper_1wepg",
    "awsui_label_1p2cx",
    "awsui_list-item_10ipo",
    "awsui_list-item_gfwv3",
    "awsui_list_1kjc7",
    "awsui_marker--highlighted_1kjc7",
    "awsui_marker_1kjc7",
    "awsui_option_1p2cx",
    "awsui_options-list_19gcf",
    "awsui_placeholder_18eso",
    "awsui_recovery_vrgzu",
    "awsui_root_10m3l",
    "awsui_root_11n0s",
    "awsui_root_1fcus",
    "awsui_root_1kjc7",
    "awsui_root_1t44z",
    "awsui_root_qwoo0",
    "awsui_root_vrgzu",
    "awsui_select_10m3l",
    "awsui_selectable-item_15o6u",
    "awsui_selected_15o6u",
    "awsui_tag_1p2cx",
    "awsui_ticks--x_f0fot",
    "awsui_ticks--y_f0fot",
    "awsui_title_1kjc7",
    "awsui_toggle_gfwv3",
    "awsui_token-content_10m3l",
    "awsui_value_10ipo",
    "awsui_wrapper_1wepg",
  ],
  "line-chart": Array [
    "awsui_root_g9v3m",
  ],
  "link": Array [
    "awsui_link_4c84z",
  ],
  "mixed-line-bar-chart": Array [
    "awsui_bar-group_1yz9c",
    "awsui_root_1yz9c",
    "awsui_series--highlighted_1yz9c",
    "awsui_series_1yz9c",
  ],
  "modal": Array [
    "awsui_content_1d2i7",
    "awsui_dismiss-control_1d2i7",
    "awsui_footer_1d2i7",
    "awsui_header_1d2i7",
    "awsui_root_1d2i7",
  ],
  "multiselect": Array [
    "awsui_root_f8g6q",
  ],
  "pagination": Array [
    "awsui_button-current_fvjdu",
    "awsui_button_fvjdu",
    "awsui_page-number_fvjdu",
    "awsui_root_fvjdu",
  ],
  "pie-chart": Array [
    "awsui_inner-content_1edmh",
    "awsui_label--highlighted_1edmh",
    "awsui_label_1edmh",
    "awsui_root_1edmh",
    "awsui_segment--highlighted_1edmh",
    "awsui_segment_1edmh",
  ],
  "popover": Array [
    "awsui_content_xjuzf",
    "awsui_dismiss-control_xjuzf",
    "awsui_header_xjuzf",
    "awsui_root_xjuzf",
    "awsui_trigger_xjuzf",
  ],
  "progress-bar": Array [
    "awsui_percentage_11huc",
    "awsui_result-button_11huc",
    "awsui_result-container-error_11huc",
    "awsui_result-container-success_11huc",
    "awsui_result-text_11huc",
    "awsui_root_11huc",
  ],
  "property-filter": Array [
    "awsui_custom-control_1wzqe",
    "awsui_custom-filter-actions_1wzqe",
    "awsui_remove-all_1wzqe",
    "awsui_root_1wzqe",
  ],
  "radio-group": Array [
    "awsui_radio_1mabk",
    "awsui_root_1mabk",
  ],
  "s3-resource-selector": Array [
    "awsui_alert_1u0yw",
    "awsui_browse-button_1vtzr",
    "awsui_layout-uri_1vtzr",
    "awsui_layout-version_1vtzr",
    "awsui_root_1u0yw",
    "awsui_root_1vtzr",
    "awsui_submit-button_12hyz",
    "awsui_view-button_1vtzr",
  ],
  "segmented-control": Array [
    "awsui_root_8cbea",
    "awsui_segment_8cbea",
    "awsui_selected_8cbea",
  ],
  "select": Array [
    "awsui_placeholder_dwuol",
    "awsui_root_r2vco",
  ],
  "side-navigation": Array [
    "awsui_divider_l0dv0",
    "awsui_expandable-link-group_l0dv0",
    "awsui_header-link_l0dv0",
    "awsui_header_l0dv0",
    "awsui_link-active_l0dv0",
    "awsui_link_l0dv0",
    "awsui_list-variant-root_l0dv0",
    "awsui_root_l0dv0",
    "awsui_section-group-title_l0dv0",
    "awsui_section-group_l0dv0",
    "awsui_section_l0dv0",
  ],
  "space-between": Array [
    "awsui_root_18582",
  ],
  "spinner": Array [
    "awsui_root_1612d",
  ],
  "split-panel": Array [
    "awsui_close-button_1r9lg",
    "awsui_drawer-closed_1r9lg",
    "awsui_header-text_1r9lg",
    "awsui_open-button_1r9lg",
    "awsui_position-bottom_1r9lg",
    "awsui_position-side_1r9lg",
    "awsui_preferences-button_1r9lg",
    "awsui_root_1r9lg",
    "awsui_slider_1r9lg",
  ],
  "status-indicator": Array [
    "awsui_root_1cbgc",
  ],
  "table": Array [
    "awsui_body-cell-edit-active_c6tup",
    "awsui_body-cell-editor-controls_c6tup",
    "awsui_body-cell-editor_c6tup",
    "awsui_body-cell_c6tup",
    "awsui_empty_wih1l",
    "awsui_expand-toggle_1ss49",
    "awsui_header-cell-ascending_1spae",
    "awsui_header-cell-descending_1spae",
    "awsui_header-controls_wih1l",
    "awsui_loading_wih1l",
    "awsui_resizer_x7peu",
    "awsui_root_1s55x",
    "awsui_root_wih1l",
    "awsui_row-selected_wih1l",
    "awsui_row_wih1l",
    "awsui_table_wih1l",
    "awsui_thead-active_wih1l",
    "awsui_tools-filtering_wih1l",
    "awsui_tools-pagination_wih1l",
    "awsui_tools-preferences_wih1l",
    "awsui_wrapper_wih1l",
  ],
  "tabs": Array [
    "awsui_root_14rmt",
    "awsui_tabs-content-active_14rmt",
    "awsui_tabs-tab-active_14rmt",
    "awsui_tabs-tab-link_14rmt",
    "awsui_tabs-tab_14rmt",
  ],
  "tag-editor": Array [
    "awsui_loading_brdst",
    "awsui_root_brdst",
    "awsui_undo-button_brdst",
  ],
  "text-content": Array [
    "awsui_text-content_6absk",
  ],
  "text-filter": Array [
    "awsui_input_1sdq3",
    "awsui_results_1sdq3",
    "awsui_root_1sdq3",
  ],
  "textarea": Array [
    "awsui_root_qk1j1",
    "awsui_textarea_qk1j1",
  ],
  "tiles": Array [
    "awsui_image_vj6p7",
    "awsui_root_vj6p7",
    "awsui_tile-container_vj6p7",
  ],
  "time-input": Array [
    "awsui_root_l809c",
  ],
  "toggle": Array [
    "awsui_root_4yi2u",
  ],
  "token-group": Array [
    "awsui_dismiss-button_dm8gx",
    "awsui_root_dm8gx",
    "awsui_token_dm8gx",
  ],
  "top-navigation": Array [
    "awsui_hidden_1ca1i",
    "awsui_hidden_k5dlb",
    "awsui_identity_1ca1i",
    "awsui_identity_k5dlb",
    "awsui_logo_1ca1i",
    "awsui_logo_k5dlb",
    "awsui_overflow-menu-back-button_k5dlb",
    "awsui_overflow-menu-dismiss-button_k5dlb",
    "awsui_overflow-menu-drawer_k5dlb",
    "awsui_overflow-menu-header-text--secondary_k5dlb",
    "awsui_overflow-menu-header-text--title_k5dlb",
    "awsui_search_1ca1i",
    "awsui_search_k5dlb",
    "awsui_title_1ca1i",
    "awsui_title_k5dlb",
    "awsui_top-navigation_1ca1i",
    "awsui_top-navigation_k5dlb",
    "awsui_trigger_1ca1i",
    "awsui_utility-wrapper_1ca1i",
    "awsui_utility-wrapper_k5dlb",
  ],
  "tutorial-panel": Array [
    "awsui_collapse-button_ig8mp",
    "awsui_completed_ig8mp",
    "awsui_completion-screen-description_1u70p",
    "awsui_completion-screen-title_1u70p",
    "awsui_dismiss-button_1u70p",
    "awsui_download-link_ig8mp",
    "awsui_expand-button_ig8mp",
    "awsui_expandable-section-wrapper_1u70p",
    "awsui_feedback-link_1u70p",
    "awsui_learn-more-link_ig8mp",
    "awsui_prerequisites-alert_ig8mp",
    "awsui_start_ig8mp",
    "awsui_step-title_1u70p",
    "awsui_task-title_1u70p",
    "awsui_task_1u70p",
    "awsui_title_ig8mp",
    "awsui_tutorial-box_ig8mp",
    "awsui_tutorial-description_ig8mp",
    "awsui_tutorial-panel_1oin5",
  ],
  "wizard": Array [
    "awsui_cancel-button_1xupv",
    "awsui_form-header-component_1xupv",
    "awsui_form-header_1xupv",
    "awsui_navigation-link-active_1xupv",
    "awsui_navigation-link-disabled_1xupv",
    "awsui_navigation-link-item_1xupv",
    "awsui_navigation-link-label_1xupv",
    "awsui_navigation-link_1xupv",
    "awsui_previous-button_1xupv",
    "awsui_primary-button_1xupv",
    "awsui_root_1xupv",
    "awsui_skip-to-button_1xupv",
  ],
}
`;
